C++/include/algorithm/sorting/bucket_sort.hpp

```diff
@@ -0,0 +1,130 @@
+/*
+    Bucket / Bin sort
+    ----------------------
+    Bucket sort, or bin sort, is an integer sorting algorithm that works by
+    distributing the elements of an array into a number of buckets. Each 
+    is then sorted individually using a different sorting algorithm.
+
+    This is a linear sorting algorithm for evenly distributed items.
+
+    Time complexity
+    ---------------
+    O(N ^ 2) worst case
+    O(N + N^2 / k + k) average case when k is the number of buckets, 
+    which is O(n) if K ~ N
+
+    Space complexity
+    ----------------
+    O(N)
+*/
+
+#ifndef BUCKET_SORT_HPP
+#define BUCKET_SORT_HPP
+
+#include <vector>
+#include <list>
+#include <algorithm>
+#include "utils.hpp"
+
+using std::vector;
+using std::list;
+using std::sort;
+
+using SortingFunction = void(*)(vector<int>&, int, bool);
+
+//Bucket sort uses another internal sorting algorithm
```

Comment:
Comments should be `// Comment`. Note the space and please fix elsewhere.

And on this line, should point out that this is using the STL sorting algorithm internally.